File: backend/src/main/java/org/dronedudes/backend/Part/Part.java
package org.dronedudes.backend.Part;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dronedudes.backend.common.Item;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.AllArgsConstructor;
import org.dronedudes.backend.Blueprint.Blueprint;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "parts")
@PrimaryKeyJoinColumn(name = "item_id")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Part extends Item {

   private String description;
   private String specifications;
   private String supplierDetails;
   private long price;


   @ManyToMany(fetch = FetchType.LAZY,
           cascade = {
               CascadeType.PERSIST,
               CascadeType.MERGE
           },
           mappedBy = "parts")
   @JsonIgnore
   private List<Blueprint> blueprints = new ArrayList<>();

   public Part(String name, String description, String specifications, String supplierDetails, long price) {
      this.setName(name);
      this.description = description;
      this.specifications = specifications;
      this.supplierDetails = supplierDetails;
      this.price = price;
   }
}

File: frontend/src/app/warehouse/_models/Part.ts
import { Item } from "./Item";

export class Part implements Item{
    id: number;
    name: string;
    type: string;
    description: string;
    specifications: string;
    supplierDetails: string;
    price: number;
    blueprints: Object[];

     constructor (id: number, name: string, description: string, specifications: string, supplierDetails: string, price: number, blueprints: Object[], type: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.specifications = specifications;
        this.supplierDetails = supplierDetails;
        this.price = price;
        this.blueprints = blueprints;
        this.type = type;
     }

}